iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusBreakerVoltageLevel.java

```diff
@@ -254,7 +254,8 @@ private void updateCache() {
                     }, encountered);
                     if (isBusValid(busSet)) {
                         String mergedBusId = BusBreakerVoltageLevel.this.id + "_" + busNum++;
-                        MergedBus mergedBus = new MergedBus(mergedBusId, busSet);
+                        String mergedBusName = BusBreakerVoltageLevel.this.name != null ? BusBreakerVoltageLevel.this.name + "_" + busNum++ : null;
```

Comment:
It seems you increment the busNum twice: the first time to generate the busId, the second time for the busName.
I think you should use busNum without increment there.

Reply:
Fixed!